src/components/ds/button/icon-button.tsx

```diff
@@ -0,0 +1,27 @@
+import styles from "./button.module.css";
+import { Loading } from "../../icons/loading";
+import classNames from "classnames/bind";
+import React, { ButtonHTMLAttributes } from "react";
+const cx = classNames.bind(styles);
+
+export interface ButtonIconProps
+  extends ButtonHTMLAttributes<HTMLButtonElement> {
+  intent?: "primary" | "secondary" | "success" | "error";
+  icon: JSX.Element;
+  loading?: boolean;
+}
+const IconButton: React.FC<ButtonIconProps> = ({
+  className,
+  intent = "primary",
+  icon,
+  loading,
+  ...props
+}) => {
+  return (
+    <button {...props} className={cx("iconButton", intent, className)}>
```

Comment:
I noticed you were putting custom classes in the ds styles. I assume its because bind cant find the styles you passed in. In that case you can separate bind from the custom className like this:
```suggestion
    <button {...props} className={classNames(cx("iconButton", intent), className)}>
```